server/src/main/java/io/druid/guice/DruidProcessingModule.java

```diff
@@ -98,6 +100,21 @@ public ExecutorService getProcessingExecutorService(
   }
 
   @Provides
+  @QueryResourceWaitingPool
+  @LazySingleton
+  public ListeningExecutorService getQueryResourceWaitingPool() throws Exception
+  {
+    return MoreExecutors.listeningDecorator(
```

Comment:
Is there a reason listeningDecorator(..) is needed and ExecutorService is not enough?